localization/pose_initializer/src/pose_initializer/pose_initializer_core.cpp

```diff
@@ -147,25 +149,52 @@ void PoseInitializer::on_initialize(
       Initialize::Service::Response::ERROR_UNSAFE, "The vehicle is not stopped.");
   }
   try {
-    change_state(State::Message::INITIALIZING);
-    change_node_trigger(false, false);
-
-    auto pose = req->pose.empty() ? get_gnss_pose() : req->pose.front();
-    if (ndt_) {
-      pose = ndt_->align_pose(pose);
-    } else if (yabloc_) {
-      // If both the NDT and YabLoc initializer are enabled, prioritize NDT as it offers more
-      // accuracy pose.
-      pose = yabloc_->align_pose(pose);
-    }
-    pose.pose.covariance = output_pose_covariance_;
-    pub_reset_->publish(pose);
+    if (req->method == Initialize::Service::Request::AUTO) {
+      change_state(State::Message::INITIALIZING);
+      change_node_trigger(false, false);
+
+      auto pose =
+        req->pose_with_covariance.empty() ? get_gnss_pose() : req->pose_with_covariance.front();
+      if (ndt_) {
+        pose = ndt_->align_pose(pose);
+      } else if (yabloc_) {
+        // If both the NDT and YabLoc initializer are enabled, prioritize NDT as it offers more
+        // accuracy pose.
+        pose = yabloc_->align_pose(pose);
+      }
+      pose.pose.covariance = output_pose_covariance_;
+      pub_reset_->publish(pose);
+
+      change_node_trigger(true, false);
+      res->status.success = true;
+      change_state(State::Message::INITIALIZED);
+
+    } else if (req->method == Initialize::Service::Request::DIRECT) {
+      if (req->pose_with_covariance.empty()) {
+        std::stringstream message;
+        message << "No input pose_with_covariance. If you wanna use DIRECT method, please input "
```

Comment:
I think it's better to keep expression more formal for error messages.

```suggestion
        message << "No input pose_with_covariance. If you want to use DIRECT method, please input "
```